TODO/my-least-favorite-thing-about-swift.md

```diff
@@ -34,17 +42,21 @@ At first blush, this looks like a language feature, like Swift’s `let` and `va
       end
     end
 
-
-
 If you can’t read Ruby, that’s okay. It uses a function called `define_method` to create a getter and setter for the keys that you pass in. In Ruby, `@first_name` means the instance variable named `first_name`.
 
+如果你不能读懂 Ruby ，没有关系。它使用了一个名为 `define_method` 的函数来为你所传递的 keys 创建一个 getter 和 setter。在 Ruby ，`@first_name` 意味着一个名为 `first_name` 的实例变量。
+
 This is one of the reasons I love Ruby’s language design — they first create the meta-tools to create useful language features, and then they use those tools to implement the language features that they want. [Yehuda Katz explores](http://yehudakatz.com/2010/02/07/the-building-blocks-of-ruby/) how Ruby applies this idea to its blocks. Because Ruby’s language features are written with the same tools and in the same language that users have access to, users can also write features similar in style and scope to the ones that define the language.
 
+这是我爱上 Ruby 这门语言的设计的其中一个原因 - 它们首先创建一个可创建有用语言特性的元数据工具集，然后它们使用这些工具来实现它们所需要的语言特性。[Yehuda Katz explores](http://yehudakatz.com/2010/02/07/the-building-blocks-of-ruby/) 讲述了 Ruby 是怎样在它的 blocks 中实现这个想法的。因为编写 Ruby 语言特性的工具和在同一门语言中用户可以访问的工具是相同的，这意味着用户也可以使用相同的风格和范围来编写定义语言的特性。
```

Comment:
Because Ruby’s language features are written with the same tools and in the same language that users have access to，users can also write features similar in style and scope to the ones that define the language
因为 Ruby 的语言特性是通过相同的工具以及相同的用户可使用的语言编写而成，所以用户也可以使用定义这门语言所相似的风格和范畴来编写语言特性。